server/src/test/java/org/elasticsearch/indices/mapping/DedicatedMasterGetFieldMappingIT.java

```diff
@@ -20,21 +20,18 @@
 package org.elasticsearch.indices.mapping;
 
 import org.elasticsearch.common.settings.Settings;
-import org.elasticsearch.node.Node;
 import org.junit.Before;
 
 import static org.elasticsearch.test.ESIntegTestCase.ClusterScope;
 import static org.elasticsearch.test.ESIntegTestCase.Scope;
+import static org.elasticsearch.test.NodeRoles.nonDataNode;
 
 @ClusterScope(scope = Scope.TEST, numDataNodes = 0)
 public class DedicatedMasterGetFieldMappingIT extends SimpleGetFieldMappingsIT {
 
     @Before
     public void before1() throws Exception {
-        Settings settings = Settings.builder()
-                .put(Node.NODE_DATA_SETTING.getKey(), false)
-                .build();
-        internalCluster().startNodes(settings, Settings.EMPTY);
+        internalCluster().startNodes(nonDataNode(), Settings.EMPTY);
```

Comment:
I see that this change is reflecting what the test was doing previously, but I also notice that the class name suggests that the node should only have the `master` role. Am I understanding it correctly, and is it worth changing the test?

Reply:
This test is *very* old though, from when we only had two roles: `data` and `master`. Historically then, disabling `data` meant the node only had the `master` role. So yes, if we want to preserve the historical intent of this test, it should be a master-only node, although functionally for this test it doesn't matter either way (if the node has the `ingest` and `remote_cluster_client` roles or not).